pennylane/operation.py

```diff
@@ -2283,9 +2293,15 @@ def is_trainable(obj):
 @qml.BooleanFn
 def defines_diagonalizing_gates(obj):
```

Comment:
Regarding the registration system: I think these functions define the problem to be solved :) The proposed registration system should allow us to delete these functions.

Reply:
Yes!